### ISSUE TITLE
Enhance the #dotParse directive to allow inclusion of a block cache (#dotcache) TTL

### ISSUE THREAD

**Issue by john-thomas-dotcms** [dotCMS/core#29144]
## Use Case
As a developer, I want to be able to easily set up my VTL code to use a block cache, without having to specifically add the `#dotcache` directive to every VTL file, or to wrap each `#dotParse` call with a `#dotcache` block.

## Details
* The #dotcache directive accepts 2 arguments: A string (to identify the dotcache "region") and a TTL.
    * To implement a block cache within the `#dotParse` directive, the string isn't needed.  Instead, we can automatically create a string by concatenating one or more of the following together:
        * The URL of the page the VTL code is being rendered on
        * The identifier of the object (container, template, content, etc.) that contains the `#dotParse` call.
        * The identifier of the VTL file being included by the `#dotParse` call.
* So, we can extend the `#dotParse` directive to just take a TTL parameter.
    * The TTL parameter should be optional.
    * If the TTL parameter is not supplied, `#dotParse` should work as it always has (without using any block cache, unless it's manually enclosed in one by the developer).

### Acceptance Criteria

1. When a `#dotParse` directive is used *without* the TTL, and *not* within a `#dotcache` block, the contents are updated based on the value of the Page TTL.
2. When a `#dotParse` directive is used *without* the TTL, and *is* within a `#dotcache` block, the contents are updated based on the value TTL supplied in the `#dotcache` block.
3. When a `#dotParse` directive is used *with* the TTL, the contents are updated based on the value TTL supplied in the `#dotParse` directive.
    * This is true whether or not the `#dotParse` block is within a `#dotcache` block or not.
4. If 2 or more `#dotParse` directives in a row use different TTLs, or if some use the TTL and some don't, the code in each `#dotParse` directive is updated based on their own TTLs.
    * In other words, the TTL from a `#dotParse` directive should never affect any code other than the code inserted by that `#dotParse` directive.

### Proposed Objective

Technical User Experience

### Proposed Priority

Priority 3 - Average

**Comment by freddyDOTCMS** [comment]
Step to Test

- Create a vtl File with the follow code:

Hello this is a test

called it test.vtl

- Create a new Widget Cotent Type, with the follow code
#dotParse('/test.vtl', 300)

- Create a Page, creta a new widget and add it to the page.
- Update the vtl FIle and change the code to: 'Hello this is a test Updated'
- Go back to the edit Page portlet and the code showed in the page must be the same
- wait 5 or 6 minutes and render the page again you must see  'Hello this is a test Updated' now

**Comment by josemejias11** [comment]
**Approved**: Tested on **trunk_36f72e4**, Docker, macOS 14.5, FF v126.0.1
